src/class/number.rs

```diff
@@ -1407,14 +1407,72 @@ where
     }
 }
 
+// Fallback trait for nb_add
+trait PyNumberAddFallback {
```

Comment:
Why `PyNumberAddFallback` instead of `PyNumberRAdd` ? (Same for the rest of the new traits & functions.)

Reply:
There's `PyNumberRAddProtocol` and `PyNumberRAddProtocolImpl` so we need a one that is clearly different from them.
We can change the implementation to use only `PyNumberRAddProtocol`, but it would need many changes in our proc-macro code. So I chose to use two traits for this, though it needs many boilerplate codes.
I don't think `RAdd` and `AddFallback` pair is not so bad, but if you come up with another idea, please let me know.